atomic_reactor/plugins/post_koji_import.py

```diff
@@ -573,14 +590,17 @@ def run(self):
                                task_state)
                 return
 
-        try:
-            for output in output_files:
-                if output.file:
-                    self.upload_file(self.session, output, server_dir)
-        finally:
-            for output in output_files:
-                if output.file:
-                    output.file.close()
+        server_dir = self.get_server_dir()
+        koji_metadata = self.combine_metadata_fragments()
+
+        if is_scratch_build(self.workflow):
+            self.upload_scratch_metadata(koji_metadata, server_dir)
+            return
```

Comment:
Previously, the plugin did this before checking the task state. I don't know if that is important, but maybe we want to keep it that way?